com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs

```diff
@@ -229,11 +233,17 @@ UnityInputProto Initialize(int port, UnityOutputProto unityOutput, out UnityInpu
             if (result.Header.Status != 200 || inputMessage.Header.Status != 200)
             {
                 m_IsOpen = false;
-                QuitCommandReceived?.Invoke();
+                NotifyQuitAndShutDownChannel();
```

Comment:
These changes doesn't look like M1 specific. Is there a bug with closing the channel?